x-pack/plugins/apm/public/components/app/ErrorGroupDetails/index.tsx

```diff
@@ -63,7 +64,8 @@ function getShortGroupId(errorGroupId?: string) {
 export function ErrorGroupDetails() {
   const location = useLocation();
   const { urlParams } = useUrlParams();
-  const { serviceName, start, end, errorGroupId, kuery } = urlParams;
+  const { uiFilters, uiFiltersKey } = useUiFilters();
+  const { serviceName, start, end, errorGroupId } = urlParams;
```

Comment:
Just being the devil's advocate while changing these things are easier:

1) Are we sure we need `useUiFilters` and don't want to do `getUiFilters(urlParams)`?

2) Are we sure we need `uiFiltersKey` for caching? Will the unhashed object not work as long as the reference is the same (so as to not to break memoization)?

Reply:
1. I think having it be a hook still makes sense based on the discussions that followed this comment (https://github.com/elastic/kibana/pull/35778#discussion_r280032012). Right now ui filters are persisted entirely in the URL, but it's likely we'll want to persist them in other ways, like localStorage.

2. The use of a string was required here since `useFetcher` defines the types of the dependency list as `useEffectKey: Array<string | boolean | number | undefined>`. If we can relax this type restriction, then we could allow a reference to be passed here.

Reply:
>  it's likely we'll want to persist them in other ways, like localStorage.

I think localStorage will never replace a url param but just be a fall back value if nothing is given in the url. We want to allow people to drill down to a specific view, and save the link (or send it to a colleague). If stuff is primarily stored in localstorage this will not be possible.

> If we can relax this type restriction, then we could allow a reference to be passed here.

Yes, I think we can relax this restriction. I probably made it because I thought that only primitive types could be part of the key. But I've come to understand that also complex types like object, arrays, functions etc. can be part of the key as long as their reference is not redefined on every render. 

If we can verify that that is the case, then we don't need `uiFiltersKey`. 

Btw. Sorry for sending you on this goose chase back and forth - I think what we have is good enough to ship but if we have time to improve and simplify unneeded details that would be great.

Reply:
to address #1: i removed the custom hook `userUiFilters` and replaced it with `getUiFilters` function which takes `urlParams` the returned value is memoized.

to address #2: i modified the dependency list type for `useFetcher` to `Array<any>` and got rid of the object hash